src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -375,6 +375,14 @@ export function getDiscoverStateContainer({
    * state containers initializing and subscribing to changes triggering e.g. data fetching
    */
   const initializeAndSync = () => {
+    const timefilerUnsubscribe = services.timefilter.getTimeUpdate$().subscribe(() => {
```

Comment:
ℹ️ This needs to be the first thing that's wired up because the next line is pulling the current state from the URL which might change the time filter and thus needs to re-check whether the saved search has changed